### ISSUE TITLE
Reconnecting...

### ISSUE THREAD

**Issue by AgentAgrimar** [SteamRE/DepotDownloader#16]
I'm trying to download a specific depot, but all it ever says is

"Reconnecting"

and trying to many times caused me to be locked out of my steam account...

**Comment by psychonic** [comment]
Can you please tell what version of DepotDownloader you're running and give the full console output?

**Comment by AgentAgrimar** [comment]
I'm using the latest release found on the GitHub.

The console output gets spammed with "Reconnecting" and eventually fails.

I kept trying it again, unaware that Steam has a login failure limit.

**Comment by mingrui** [comment]
Hi I'm having the same reconnecting problem
I was only able to login the first time
tried about 10 times after the first successful login and all failed.

**Comment by azuisleet** [comment]
Can you post command lines (without password)? I'm using the 2.2.1 release from the zip and I do not experience anything usual with authenticated or anonymous downloads.

**Comment by mingrui** [comment]
Thanks for the quick reply!
I tried again on ubuntu 16.04 after rechecking my installation of mono according to this thread
https://github.com/SteamRE/DepotDownloader/issues/12

this is my command lines:
```
mono DepotDownloader.exe -app 250820 -depot 250823 -manifest 9183475244149741279 -username mjray369
Enter account password for "mjray369": 
Connecting to Steam3... Done!
Logging 'mjray369' into Steam3...This account is protected by Steam Guard.
Please enter your 2 factor auth code from your authenticator app:
```

after I enter my steam guard code, there is an endless stream of "Reconnecting" ending this:

```
Reconnecting
Timeout connecting to Steam3.
Unable to get steam3 credentials.
```

**Comment by azuisleet** [comment]
I was able to reproduce this. I have a fix pending in 35aa666 that needs some review. The Disconnected callback should have been ready on the callback queue, but we weren't processing it until after our next Connect (and thus the cycle repeated itself).

I need to review when the Disconnected callback is actually appearing now, since it should be seen immediately upon processing the queue.

**Comment by mingrui** [comment]
Hi azuisleet,

Is there an older .exe version I can use? Or should I try to build the .exe myself using your update?
I really need that older version of steamvr for linux fast because I have a tool that's broken by the latest update.

Do you know any other way to download a previous version of steamvr?

Thanks for your help!

**Comment by azuisleet** [comment]
This has been fixed and released in DepotDownloader 2.2.2
